pkg/catalog/routes.go

```diff
@@ -333,12 +335,31 @@ func getTrafficPoliciesForService(mc *MeshCatalog, routePolicies map[trafficpoli
 		}
 	}
 
-	for _, trafficTarget := range srcDstTrafficTargetMap {
+	/*for _, trafficTarget := range srcDstTrafficTargetMap {
 		matchedTrafficTargets = append(matchedTrafficTargets, trafficTarget)
 	}
 
 	log.Debug().Msgf("Traffic policies for service %s: %+v", meshService, matchedTrafficTargets)
-	return matchedTrafficTargets, nil
+	return matchedTrafficTargets, nil*/
+
+	return srcDstTrafficTargetMap, nil
+}
+
+func (mc *MeshCatalog) getTrafficRoutesForService(svc service.MeshService, trafficTargets map[string]trafficpolicy.TrafficTarget) []trafficpolicy.TrafficRoutes {
+	trafficRoutes := []trafficpolicy.TrafficRoutes{}
+
+	for _, target := range trafficTargets {
+		trafficRoute := trafficpolicy.TrafficRoutes{
+			//Name:                  target.Name,
+			Source:                target.Source,
+			Destination:           target.Destination,
+			RouteWeightedServices: mc.getRouteWeightedServices(target.Source, target.Destination, trafficTargets),
+		}
+		trafficRoutes = append(trafficRoutes, trafficRoute)
+	}
+
+	//TODO: Add roots of traffic splits (will need some slight testing changes)
```

Comment:
Discussed offline to get all unique apex services and then build the traffic routes for these apex services